### ISSUE TITLE
 CloudFormation cannot update a stack when a custom-named resource requires replacing when using @key

### ISSUE THREAD

**Issue by sprucify** [aws-amplify/amplify-cli#1651]
Can I use @key on an existing table with data?
Because when I add a @key field to my model I get:
` CloudFormation cannot update a stack when a custom-named resource requires replacing.`
and the CloudFormation is rolling back.

The table has over 2k items of data in it, so dropping the table is not a good option

**Comment by yuth** [comment]
@sprucify could you share the Model before and after the change

**Comment by warrenmcquinn** [comment]
Just chiming into say that I ran into this issue as well.

Here's the model pre-key:
```
type List
  @model
{
  id: ID!
  owners: [String]!
  name: String
  domain: String
  subdomain: String!
}
```

and after:
```
type List
  @model
  @key(fields: ["subdomain"])
{
  id: ID!
  owners: [String]!
  name: String
  domain: String
  subdomain: String!
}
```

CF rollback occurred. Thankfully this is on a prototype, so I'm able to delete the resource and redeploy, but I would hope there's a way forward for others who have nonfungible DBs.

**Comment by corwinm** [comment]
Also just ran into this. Not a big deal for my situation but interested where this ends up going.

**Comment by sprucify** [comment]
My pre-key model with table that contained data:
```
type Recipe 
  @model
{
  id: ID!
  title: String
  key: String
  image: String
  numberOfPersons: Int
  courses: [String]
  time: String
  energy: String
  proteine: String
  carbohydrates: String
  fat: String
  saturatedFat: String
  fibers: String
  salt: String
  sugar: String
  recipeTypes: [String]
  specials: [String]
  containsProducts: [String]
  diet: [String]
  ingredients: [Ingredient]
  preperation: [String]
  editDate: String
  menuSuggestion: String
  weeknumber: Int!
}
```

after 

```
type Recipe 
  @model @key(fields: ["weeknumber"])
{
  id: ID!
  title: String
  key: String
  image: String
  numberOfPersons: Int
  courses: [String]
  time: String
  energy: String
  proteine: String
  carbohydrates: String
  fat: String
  saturatedFat: String
  fibers: String
  salt: String
  sugar: String
  recipeTypes: [String]
  specials: [String]
  containsProducts: [String]
  diet: [String]
  ingredients: [Ingredient]
  preperation: [String]
  editDate: String
  menuSuggestion: String
  weeknumber: Int!
}
```

**Comment by sudilshr** [comment]
Having same issue after adding @key on existing table.

**Comment by yuth** [comment]
When `@key` directive is used with out the name argument, GraphQL transformer treats it as a primary index. For an existing model which did not use any `@key` directive, transformer would have already created the primary index.

If you're editing a model which is already pushed, I would recommend you give a name to your `@key` directive as updating the primary index in Dynamo DB is not supported

**Comment by sudilshr** [comment]
Thanks @yuth. My issue is resolved after providing name to @key.

**Comment by yuth** [comment]
Please feel free to comment/re-open issue if you're still facing this problem

**Comment by mart2967** [comment]
@yuth I'm hitting this issue when adding the first key to my schema, like so:

`@key(name: "ByPlays", fields: ["id", "plays"], queryField: "songsByPlays")`

**Comment by khusmann** [comment]
@yuth I am having the same issue as @mart2967 : 

Add to model:

@key(name: "ByStartDate", fields: ["id", "startDate"], queryField: "activitySessionsByStartDate")

Error: 

Resource Name: ActivitySession-*** (AWS::DynamoDB::Table)
Event Type: update
Reason: CloudFormation cannot update a stack when a custom-named resource requires replacing. Rename ActivitySession-***-master and update the stack again.

Thanks!

**Comment by ilyador** [comment]
Having the same issue, even with `name` parameter
```
type Set @model @key(name: "ByAppearedForRanking", fields: ["appearedForRanking"]) {
  id: ID!
  user: String!
  appearedForRanking: Int!
  pictures: [Picture!]! @connection
}
```

**Comment by dylan-westbury** [comment]
I updated the key params and am experiencing the same error as above.

> Resource Name: Ticket-ffwb6tn4yzezllhzc4dgxeofpa (AWS::DynamoDB::Table)
Event Type: update
Reason: CloudFormation cannot update a stack when a custom-named resource requires replacing. Rename Ticket-ffwb6tn4yzezllhzc4dgxeofpa-dev and update the stack again.


I tried deleting the dynamodb tables that are throwing the rename error, but it is still telling me to rename them even though they don't exist anymore...
